spec/validations/svg_spec.rb

```diff
@@ -96,99 +92,116 @@
+
+        errors = ZendeskAppsSupport::Validations::Svg.call(package)
 
-    dirty_markup_examples.map do |markup|
-      let(:markup) { markup }
-      let(:clean_markup) do
-        %(<svg viewBox="0 0 26 26" id="zd-svg-icon-26-app" width="100%" height="100%"><path fill="none" \
+        clean_markup = %(<svg viewBox="0 0 26 26" id="zd-svg-icon-26-app" width="100%" height="100%"><path fill="none" \
 stroke="currentColor" stroke-linejoin="round" stroke-width="2" d="M4 8l9-5 9 5v9.7L13 23l-9-5.2zm9 5L4 8m9 \
 5l9-5m-9 5v10"/></svg>)
-      end
-      it 'are sanitised, with a warning that the offending svgs were modified' do
-        errors = subject.call(package)
-        expect(IO).to have_received(:write).with(svg.absolute_path, clean_markup)
-        expect(package.warnings[0]).to eq(warning)
+        expect(IO).to have_received(:write).with('/tmp/icon.svg', clean_markup)
+        expect(package.warnings[0]).to eq(
+          'The markup in assets/icon.svg has been edited for use in Zendesk and may not display as intended.'
+        )
         expect(errors).to be_empty
       end
     end
+
+    it 'removes <script> tags' do
+      markup = %(<svg viewBox="0 0 26 26" id="zd-svg-icon-26-app" width="100%" height="100%"><path fill="none" \
+stroke="currentColor" stroke-linejoin="round" stroke-width="2" d="M4 8l9-5 9 5v9.7L13 23l-9-5.2zm9 5L4 8m9 \
+5l9-5m-9 5v10"><script type="text/javascript">alert(1);</script></path></svg>\n)
+      svg_file = double('SVGFile', read: markup, relative_path: 'assets/icon.svg', absolute_path: '/tmp/icon.svg')
+      package = mock_package(svg_file)
+
+      errors = ZendeskAppsSupport::Validations::Svg.call(package)
+
+      clean_markup = %(<svg viewBox="0 0 26 26" id="zd-svg-icon-26-app" width="100%" height="100%"><path fill="none" \
+stroke="currentColor" stroke-linejoin="round" stroke-width="2" d="M4 8l9-5 9 5v9.7L13 23l-9-5.2zm9 5L4 8m9 \
+5l9-5m-9 5v10"></path></svg>)
+      expect(IO).to have_received(:write).with('/tmp/icon.svg', clean_markup)
+      expect(package.warnings[0]).to eq(
+        'The markup in assets/icon.svg has been edited for use in Zendesk and may not display as intended.'
+      )
+      expect(errors).to be_empty
+    end
   end
 
   context 'svgs containing malformed, suspicious markup' do
-    malformed_markup_examples = [
+    [
       %(<svg onload=innerHTML=location.hash>#<script>alert(1)</script> viewBox="0 0 26 26" id="zd-svg-icon-26-app" \
 width="100%" height="100%"><path fill="none" stroke="currentColor" stroke-linejoin="round" stroke-width="2" \
 d="M4 8l9-5 9 5v9.7L13 23l-9-5.2zm9 5L4 8m9 5l9-5m-9 5v10"><script type="text/javascript">alert('XSS');</script> \
       </path></svg>\n),
+      %(<svg onload=alert&#x28;1&#x29 viewBox="0 0 26 26" id="zd-svg-icon-26-app" width="100%" height="100%"> \
+<path fill="none" stroke="currentColor" stroke-linejoin="round" stroke-width="2" d="M4 8l9-5 9 5v9.7L13 \
+23l-9-5.2zm9 5L4 8m9 5l9-5m-9 5v10"></path></svg>\n),
```

Comment:
Why are we adding this now?

Reply:
Because this example was in the wrong group. This suite was borked and was running the last example multiple times (See https://github.com/zendesk/zendesk_apps_support/pull/349/commits/da1280098688d75306ec68af298ab87b772db79b).